code/modules/mob/living/carbon/human/species_types/jellypeople.dm

```diff
@@ -572,16 +578,17 @@
 		return
 	CHECK_DNA_AND_SPECIES(H)
 
-	if(species.current_extract)
-		species.extract_cooldown = world.time + 100
+	if(species.current_extract && H.blood_volume >= blood_cost)
 		var/cooldown = species.current_extract.activate(H, species, activation_type)
 		species.extract_cooldown = world.time + cooldown
+		H.blood_volume -= blood_cost
```

Comment:
does this mean that if you start one of the do_after() extract activations but that do_after() gets interrupted, you'll still have to pay its blood cost?

Reply:
Yep